models/auth/oauth2.go

```diff
@@ -31,6 +31,8 @@ type OAuth2Application struct {
 	Name         string
 	ClientID     string `xorm:"unique"`
 	ClientSecret string
+	// https://datatracker.ietf.org/doc/html/rfc6749#section-2.1
+	Confidential bool               `xorm:"NOT NULL DEFAULT false"`
```

Comment:
The old behaviour is Confidential=true. Is that okay for compability? The alternative would be to call the field Public so the old behaviour is Public=false.

What does the xorm default do?

Reply:
Xorm when syncing will add the column `confidential` to the table and set every value to false.

Reply:
Thanks for the explanation